File: benchmarks/cpp/recursion/recursion.cpp
#include <iostream>
#include <chrono>
using namespace std;
using namespace chrono;

long long fib(long long n) {
    if (n <= 1)
        return n;
    return fib(n - 1) + fib(n - 2);
}

int main() {
    const int N = 40;

    auto start = high_resolution_clock::now();

    volatile long long result = fib(N);

    auto end = high_resolution_clock::now();
    duration<double> elapsed = end - start;

    cout << "C++ Recursion Time: " << elapsed.count() << " seconds\n";
    return 0;
}

File: benchmarks/rust/recursion/recursion.rs
use std::time::Instant;

fn fib(n: i64) -> i64 {
    if n <= 1 {
        n
    } else {
        fib(n - 1) + fib(n - 2)
    }
}

fn main() {
    let n: i64 = 40;

    let start = Instant::now();

    let _result = fib(n);

    let elapsed = start.elapsed();
    println!("Rust Recursion Time: {:.6} seconds", elapsed.as_secs_f64());
}

